examples/common/websocket-server/WebSocketServer.cpp

```diff
@@ -166,10 +166,15 @@ CHIP_ERROR WebSocketServer::Run(chip::Optional<uint16_t> port, WebSocketServerDe
 
     lws_context_creation_info info;
     memset(&info, 0, sizeof(info));
-    info.port             = port.ValueOr(kDefaultWebSocketServerPort);
-    info.iface            = nullptr;
-    info.pt_serv_buf_size = kMaxMessageBufferLen;
-    info.protocols        = protocols;
+    info.port                         = port.ValueOr(kDefaultWebSocketServerPort);
+    info.iface                        = nullptr;
+    info.pt_serv_buf_size             = kMaxMessageBufferLen;
+    info.protocols                    = protocols;
+    static const lws_retry_bo_t retry = {
+        .secs_since_valid_ping   = 400,
+        .secs_since_valid_hangup = 400,
+    };
```

Comment:
This is not initializing all the struct members.  What values do the other members end up with, and why are those values OK?

Reply:
> This is not initializing all the struct members

It does initialize them implicitly to zero - the `retry` has static storage duration.

Reply:
@bzbarsky-apple @arkq 
Thanks for your interest!

> What values do the other members end up with, and why are those values OK?

Eventually it will look like this:
```
[D] retry_ms_table_count=4
[D] conceal_count=4
[D] secs_since_valid_ping=400
[D] secs_since_valid_hangup=400
[D] jitter_percent=20
```

I didn't make initialization of other struct members because it would be useless as in [libwebsockets/repo/lib/core/context.c#L1138](https://github.com/warmcat/libwebsockets/blob/288f36fc8178e2664614b78322d2d05a70b9ac0a/lib/core/context.c#L1138) those values are set from `default_retry` structure thus cannot be redefined as `secs_since_valid_ping` and `secs_since_valid_hangup`. 
I've even tested that manually, they don't change in runtime anywhere. 
I don't think we need to redefine all the variables in the `lws_retry_bo_t` as default are OK. We regularly run certification tests on the development board and they all pass now, no issues regarding websocket or any others.

BUT if we want to do that, then _maybe_ not the `lws_create_context` should be used in `connectedhomeip/examples/common/websocket-server/WebSocketServer.cpp` but `lws_client_connect_via_info` from `connectedhomeip/third_party/libwebsockets/repo/lib/core-net/client/connect.c`. 
 
Static initialization was taken as a referrence from repo's examples like this one in [libwebsockets/repo/minimal-examples-lowlevel/ws-client/minimal-ws-client/minimal-ws-client.c](https://github.com/warmcat/libwebsockets/blob/288f36fc8178e2664614b78322d2d05a70b9ac0a/minimal-examples-lowlevel/ws-client/minimal-ws-client/minimal-ws-client.c#L39)

Please, let me know if I should make any changes considering all above and thanks for your time!

Reply:
@interfer Ah, I see.  Ok, perfect, that's what I needed to know.